kernel/src/table_properties.rs

```diff
@@ -0,0 +1,374 @@
+    #[serde(rename = "delta.checkpointInterval")]
+    #[serde(deserialize_with = "deserialize_pos_int")]
+    pub checkpoint_interval: Option<u64>,
+
+    /// true for Delta Lake to write file statistics in checkpoints in JSON format for the stats column.
+    #[serde(rename = "delta.checkpoint.writeStatsAsJson")]
+    #[serde(deserialize_with = "deserialize_bool")]
+    pub checkpoint_write_stats_as_json: Option<bool>,
+
+    /// true for Delta Lake to write file statistics to checkpoints in struct format for the
+    /// stats_parsed column and to write partition values as a struct for partitionValues_parsed.
+    #[serde(rename = "delta.checkpoint.writeStatsAsStruct")]
+    #[serde(deserialize_with = "deserialize_bool")]
+    pub checkpoint_write_stats_as_struct: Option<bool>,
+
+    /// Whether column mapping is enabled for Delta table columns and the corresponding
+    /// Parquet columns that use different names.
+    #[serde(rename = "delta.columnMapping.mode")]
+    #[serde(deserialize_with = "deserialize_option")]
+    pub column_mapping_mode: Option<ColumnMappingMode>,
+
+    /// The number of columns for Delta Lake to collect statistics about for data skipping.
+    /// A value of -1 means to collect statistics for all columns. Updating this property does
+    /// not automatically collect statistics again; instead, it redefines the statistics schema
+    /// of the Delta table. Specifically, it changes the behavior of future statistics collection
+    /// (such as during appends and optimizations) as well as data skipping (such as ignoring column
+    /// statistics beyond this number, even when such statistics exist).
+    #[serde(rename = "delta.dataSkippingNumIndexedCols")]
+    #[serde(deserialize_with = "deserialize_option")]
+    pub data_skipping_num_indexed_cols: Option<DataSkippingNumIndexedCols>,
+
+    /// A comma-separated list of column names on which Delta Lake collects statistics to enhance
+    /// data skipping functionality. This property takes precedence over
+    /// `delta.dataSkippingNumIndexedCols`.
+    #[serde(rename = "delta.dataSkippingStatsColumns")]
+    #[serde(deserialize_with = "deserialize_column_names")]
+    pub data_skipping_stats_columns: Option<Vec<ColumnName>>,
+
+    /// The shortest duration for Delta Lake to keep logically deleted data files before deleting
+    /// them physically. This is to prevent failures in stale readers after compactions or partition
+    /// overwrites.
+    ///
+    /// This value should be large enough to ensure that:
+    ///
+    /// * It is larger than the longest possible duration of a job if you run VACUUM when there are
+    ///   concurrent readers or writers accessing the Delta table.
+    /// * If you run a streaming query that reads from the table, that query does not stop for
+    ///   longer than this value. Otherwise, the query may not be able to restart, as it must still
+    ///   read old files.
```

Comment:
aside: will we have a check that blocks too-small intervals? ie less than one day?

Reply:
we could! is that in the protocol/implement in delta-spark? (i'll check too)

Reply:
I think it's in the delta-spark impl? You have to "break glass" to vacuum with an interval less than X?

Reply:
perhaps take this as a follow-up? (not implementing vacuum currently and other implementations that consume this could check themselves?)

created https://github.com/delta-io/delta-kernel-rs/issues/522